### ISSUE TITLE
Failed builds with "unexpected EOF"

### ISSUE THREAD

**Issue by mbainter** [mondoohq/packer-plugin-cnspec#187]
**Describe the bug**
When running a build that uses the cnspec provisioner it fails for me with "unexpected EOF".  

**To Reproduce**

1. set MONDOO_CONFIG_BASE64
2. configure the cnspec provisioner in the packer build (so far happens on ubuntu focal and amazon2)
3. Run a build

**Expected behavior**
1. If the scan fails, I would expect to see errors that indicate that.
2. If I re-run the build with PACKER_LOG=1 I expect to see some extra log information that can help me understand what's going wrong
3. Ultimately I expect the scan to complete and report the results to mondoo.

**Screenshots or CLI Output**
What I actually get is this:
```
==> amazon-ebs.ami: Running cnspec packer provisioner by Mondoo (Version: 10.0.3, Build: b2dd4a6)
    amazon-ebs.ami: activated sudo
    amazon-ebs.ami: detected packer build via ssh
    amazon-ebs.ami: load config from detected MONDOO_CONFIG_BASE64
    amazon-ebs.ami: using service account credentials
    amazon-ebs.ami: scan packer build
    amazon-ebs.ami: scan completed successfully
==> amazon-ebs.ami: unexpected EOF
==> amazon-ebs.ami: Step "StepProvision" failed
```

And if I "retry" the step then packer crashes.  But, the crash output does show me some useful additional information. Note that this is from the initial attempt, not the retry:
```
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin: DBG could not parse ssh port error="strconv.Atoi: parsing \"\": invalid syntax" file=/home/mbainter/.ssh/config port=
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin: DBG ssh> read ssh identity key from ssh config host=10.15.12.92 key=
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin: DBG could not establish ssh session error="no authentication method defined" host=xx.xx.xx.xx i
nsecure=true port=22 provider=ssh
2024/01/26 15:03:15 ui: ESC[0;32m    amazon-ebs.ami: scan completed successfullyESC[0m
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin: panic: runtime error: invalid memory address or nil pointer dereference
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin: [signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x1913965]
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin:
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin: goroutine 26 [running]:
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin: go.mondoo.com/cnspec/v10/policy.(*ReportCollection).GetWorstScore(...)
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin:       /home/runner/go/pkg/mod/go.mondoo.com/cnspec/v10@v10.0.3/policy/report.go:159
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin: go.mondoo.com/packer-plugin-cnspec/provisioner.(*Provisioner).executeCnspec(0xc000005500, {0x23
977c0, 0xc000bb40c0}, {0xc000005700?, 0xc0007f0a88?})
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin:       /home/runner/_work/packer-plugin-cnspec/packer-plugin-cnspec/provisioner/provisioner.go:6
04 +0x2165
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin: go.mondoo.com/packer-plugin-cnspec/provisioner.(*Provisioner).Provision(0xc000005500, {0x3289a0
0?, 0xc00079d230?}, {0x23977c0, 0xc000bb40c0}, {0x2396208?, 0xc0007b23a0}, 0xc000a81440)
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin:       /home/runner/_work/packer-plugin-cnspec/packer-plugin-cnspec/provisioner/provisioner.go:3
01 +0x799
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin: github.com/hashicorp/packer-plugin-sdk/rpc.(*ProvisionerServer).Provision(0xc00092e040, 0xc0006
57b90, 0x1?)
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin:       /home/runner/go/pkg/mod/github.com/hashicorp/packer-plugin-sdk@v0.5.2/rpc/provisioner.go:
91 +0x1c9
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin: reflect.Value.call({0xc00080c360?, 0xc0000b6120?, 0x13?}, {0x1dfe297, 0x4}, {0xc000865ef8, 0x3,
 0x3?})
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin:       /home/runner/_work/_tool/go/1.21.6/x64/src/reflect/value.go:596 +0xce7
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin: reflect.Value.Call({0xc00080c360?, 0xc0000b6120?, 0x7fed5f9d4a88?}, {0xc0007c86f8?, 0xc0007c86b
8?, 0x4b59b0?})
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin:       /home/runner/_work/_tool/go/1.21.6/x64/src/reflect/value.go:380 +0xb9
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin: net/rpc.(*service).call(0xc00092e080, 0xc0007c87b8?, 0xb7d838?, 0xc0006d8060, 0xc000926180, 0x6
96c222c7d7d226b?, {0x19dbaa0?, 0xc000657b90?, 0xb7aea5?}, {0x1a07340, ...}, ...)
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin:       /home/runner/_work/_tool/go/1.21.6/x64/src/net/rpc/server.go:382 +0x211
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin: created by net/rpc.(*Server).ServeCodec in goroutine 1
2024/01/26 15:03:15 packer-plugin-cnspec_v10.0.3_x5.0_linux_amd64 plugin:       /home/runner/_work/_tool/go/1.21.6/x64/src/net/rpc/server.go:479 +0x410
2024/01/26 15:03:15 [INFO] (telemetry) ending cnspec
```

**Desktop (please complete the following information):**

- OS: Linux
- OS Version: PopOS
- Packer: 1.10.0
- CNSpec Plugin: 9.14 and 10.0.3

**Additional context**

I thought maybe something in my ssh config was tripping it up based on the log data, so I removed it so it wouldn't be a factor. it still failed the same way.  

I also tried running just a straight cnspec ssh to make sure my ssh setup wasn't somehow incompatible with cnspec itself, but that runs fine with `cnspec scan ssh ubuntu@<host>`

**Comment by chris-rock** [comment]
@mbainter super helpful. I was able to reproduce parts of your case. I found some edge cases where the packer plugin can trigger this behavior. So far I was only able to reproduce those with Windows + Winrm. Seems like you are testing a remote Ubuntu system. What would be helpful if you could share the build script, so that we can try to reproduce your exact case.

We are going to update the implementation to handle the case where the report is not properly generated.

**Comment by mbainter** [comment]
I'm building on PopOS, against an ubuntu target, using SSH.  @scottford-io has a tarball of a build template that fails for me, but it's slimmed down to just a shell provisioner that waits for cloud-init (which just does an apt-get update/upgrade) and the cnspec provisioner.

Since it seems like the ssh config is likely connected, as I said above I removed my user config file to ensure that's not a factor.  The only potentially unique thing about this build attempt I'm doing is that we use `ssh_agent_auth` in our packer build and then I am using gpg-agent locally to serve the key.

**Comment by chris-rock** [comment]
Thank you, I am chatting with @scottford-io then.

**Comment by chris-rock** [comment]
Quick update: We are able to reproduce an issue where `unexpected EOF` happens when ssh public key auth is used:

```hcl
source "amazon-ebs" "ubuntu2204" {
  ...

  ssh_keypair_name = "chris-rock"
  ssh_agent_auth   = true
}
```

**Comment by chris-rock** [comment]
Hi @mbainter we released v10.1.2 which includes support for ssh agent authentication and also improved the error reporting. I would appreciate if you could test the plugin in your environment to see if everything works as expected now.
